optuna/trial/_base.py

```diff
@@ -54,8 +54,8 @@ def suggest_discrete_uniform(self, name, low, high, q):
         raise NotImplementedError
 
     @abc.abstractmethod
-    def suggest_int(self, name, low, high, step=1):
-        # type: (str, int, int, int) -> int
+    def suggest_int(self, name, low, high, *, log=False, step=1):
```

Comment:
Please correct me if this has been discussed before but this seems like a breaking change for users that are calling `suggest_int` while specifying `step` as a positional argument. How about keeping `step` in the same position, add `log` afterwards, and
a. raise a deprecation warning that arguments orders will change, then change the order in the future, or
b. rearrange the order in `suggest_float` to match `suggest_int`?

Reply:
I now see that a discussed was held here https://github.com/optuna/optuna/pull/1201#discussion_r419170507, but it still seems like an issue to me.

Reply:
I agree with @hvy's idea (b). It might be better to change the order of arguments in `suggest_int` to `suggest_int(low, high, step, log)`. Changing the argument order of `suggest_float` can be a future work (another PR).

Reply:
Thank you for your suggestion. Personally, option b seems better because both arguments are already keyword-only arguments in `suggest_float`. So its order can be changed with keeping the backward compatibility,

Reply:
Thanks for the swift discussion. Sounds good. Let's deal with `suggest_float` after this PR (it'd mostly be a cosmetic fix).